supermarq-benchmarks/supermarq/qcvv/base_experiment.py

```diff
@@ -45,6 +46,19 @@ class Sample:
     """The corresponding data about the circuit that is needed when analyzing results
     (e.g. cycle depth)."""
 
+    uuid: uuid.UUID = field(default_factory=uuid.uuid1)
+    """The unique ID of the sample."""
+
+    def __hash__(self) -> int:
+        return hash(
+            (
+                self.circuit_index,
+                self.uuid,
+                cirq.to_json(self.circuit),
```

Comment:
this should work i think
```suggestion
                self.circuit.freeze(),
```